File: backend/src/main/java/org/globe42/web/security/AuthenticationConfig.java
package org.globe42.web.security;

import java.util.Arrays;

import org.springframework.boot.web.servlet.FilterRegistrationBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * Configuration of the WebMvc application
 * @author JB Nizet
 */
@Configuration
public class AuthenticationConfig {

    @Bean
    public FilterRegistrationBean<AuthenticationFilter> authenticationFilterRegistration() {
        FilterRegistrationBean<AuthenticationFilter> filterRegistrationBean =
            new FilterRegistrationBean<>(authenticationFilter());
        filterRegistrationBean.setUrlPatterns(Arrays.asList("/api/*", "/actuator", "/actuator/*"));
        return filterRegistrationBean;
    }

    @Bean
    public AuthenticationFilter authenticationFilter() {
        return new AuthenticationFilter();
    }
}

File: backend/src/main/kotlin/org/globe42/web/security/AuthenticationConfig.kt
package org.globe42.web.security

import org.globe42.dao.UserDao
import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * Configuration of the WebMvc application
 * @author JB Nizet
 */
@Configuration
class AuthenticationConfig {

    @Bean
    fun authenticationFilterRegistration(authenticationFilter: AuthenticationFilter): FilterRegistrationBean<AuthenticationFilter> {
        return FilterRegistrationBean(authenticationFilter).apply {
            urlPatterns = listOf("/api/*", "/actuator", "/actuator/*")
        }
    }

    @Bean
    fun authenticationFilter(jwtHelper: JwtHelper, currentUser: CurrentUser, userDao: UserDao) =
        AuthenticationFilter(jwtHelper, currentUser, userDao)
}
